Add opus audio encoding support

jstebbins opened the pull request:
This isn't quite finished and only lightly tested.  But I'm going to be out of touch for another week, so I thought I would post this so that others can do some testing and prepare any UI work that needs to be done.

CLI and Linux and OSX UIs should have basic support.  Some work needs to be done in the UIs to limit the samplerates available when opus is the encoder.

jstebbins committed:
Add opus audio encoding support

Djfe commented:
it would be awesome if you could use/support Opus 1.1.3 from the start :)
there have been a few fixes/optimizations since 1.1

jstebbins committed:
opus: bump version to 1.1.3

jstebbins committed:
LinGui: disable invalid audio samplerates

opus only supports a limited set of samplerates.  disable those that do
not apply.

jstebbins commented:
This could use more testing.  I only did some very limited smoke tests.  But I believe the libhb and LinGui bits are complete.

Should we commit to get wider testing started?  WinGui and MacGui can add necessary bits as time permits.

sr55 commented:
Sure, why not.

galad87 commented:
The MacGui requires some little changes (to use the new hb_audio_samplerate_find_closest function instead of hb_audio_samplerate_get_best), but it seems to work. Is the max sample rate 22.05?

Last thing, should we add an auto passthru option for Opus too?

galad87 commented:
Discard my comment on sample rate, the popup menu was not being updated correctly in the MacGui.

jstebbins commented:
Opus passthru should be a separate patch if we want to do that.  I had only marginal interest in opus encoding and even less interest in opus passthru.

sr55 commented:
Agreed. I think for now, we skip passthru as we want to wrap up this release asap.

galad87 commented:
+1 to merge this.

bradleysepos commented:
Rebased and merged as fb78eca4a49077c171ad78177c1615c205adc642 and 59803dafcc75c4a2364faf172d93ffef3985e300.
